### ISSUE TITLE
Callback after every recomputation of a Tracker computation

### ISSUE THREAD

**Issue by mitar** [meteor/meteor#5354]
Currently there is only `afterFlush` which is rerun only if computation was not invalidated while it was computed. Otherwise the computation is rerun immediately. So there is no way to register a callback to be called after every recomputation. This would help decouple content of a computation from the reactive functions inside a computation which could register callbacks to be called after that exact computation finishes.

**Comment by stubailo** [comment]
> help decouple content of a computation from the reactive functions inside a computation

Can you help me understand a little more what you mean by this? I'm not really sure how a callback on recomputation isn't the same as the computation itself.

**Comment by mitar** [comment]
Hm, OK, imagine that I am trying to create a library with some reactive functions. And user could use them inside an autorun:

```javascript
Tracker.autorun(function (computation) {
  var foo = MyLibrary.callsomething();
  doSomethingWith(foo);
  doSomeMore(foo);
  // HERE
});
```

Currently, there is no way for `callsomething` method call to inject any code to be called at the point marked with `HERE`. I can use `Tracker.afterFlush` to do something similar, but if code after `callsomething` invalidates computation again, then `afterFlush` will not be called after each recomputation, but only after all recomputations finish. This means that there could be some starvation happening or even invalid logic if the code would really have to be called after every recomputation.

Of course, I could provide some method for users to call at the end, but that becomes error prone become users may forget to add a call at the end. Or computation could finish prematurely for some code paths.

Concrete example for this is in my server-side [reactive publish](https://github.com/peerlibrary/meteor-reactive-publish) library. Currently I am using [afterFlush](https://github.com/peerlibrary/meteor-reactive-publish/blob/master/server.coffee#L53) to know check which documents were added in this particular computation run and if there are any documents which were added previously, but not this time, I can remove them. The issue is if computation gets invalidated during the run, then it gets called immediately and `afterFlush` gets delayed. And this can be repeating for quite some time under the load (so starving happens). It would be great if I could register a callback to be called after every recomputation, and not only after there is a recomputation which has not been invalidated in meantime.

**Comment by mitar** [comment]
See #4410 for additional motivation for this callback.

**Comment by mitar** [comment]
I implemented this in #5392.

**Comment by hwillson** [comment]
To help provide a more clear separation between feature requests and bugs, and to help clean up the feature request backlog, Meteor feature requests are now being managed under the https://github.com/meteor/meteor-feature-requests repository. 

This feature request will be closed here, but anyone interested in migrating this feature request to the new repository (to make sure it stays active), can click [here](https://github.com/meteor/meteor-feature-requests/issues/new?title=Callback%20after%20every%20recomputation%20of%20a%20Tracker%20computation&body=Migrated%20from:%20meteor/meteor%235354) to start the feature request migration process. This manual migration process is intended to help identify which of the older feature requests are still considered to be of value to the community. Thanks!
